qualtran/bloqs/arithmetic/sorting.ipynb

```diff
@@ -36,19 +36,20 @@
    },
    "source": [
     "## `Comparator`\n",
-    "Compare and potentially swaps two n-bit numbers.\n",
+    "Compare and potentially swaps two numbers in {0, ... L - 1}.\n",
```

Comment:
Why the change from bitsize (which is common in qualtran) to integer range. Is it just more convenient?

Reply:
papers seem to use "integers in range [n]" etc. often, so I thought this would be more useful. And having `BoundedQUInt` makes it much nicer to support this than earlier.

Reply:
Is there an advantage in terms of complexity of sorting? I would say this notation is definitely more confusing. I had to look at the unit tests to figure out what you mean by `Sort k numbers in the range {0, ..., L-1}.` My first guess was maybe you want to sort the first `k` of the first `L` numbers or something. 

Saying something like `Sort K n-bit numbers in complexity `O()` complexity` is a lot cleaner. If there's no advantage in terms of complexity and everywhere we are using `n=bitsize(L - 1)` to determine the bitsize anyway, then I'd say lets change the bloqs to use `bitsize` of the numbers to be sorted as the attribute by default and we can provide a builder method that expects an `L` and computes the bitsize for you.

Reply:
Actually, as the comparison bloqs only accept bitsize, this doesn't give any improvement. It does make more sense to just keep it as bitsize, (and provide a helper constructor that takes `L` ). I'll revert the changes and update the docs.

Reply:
Used bitsizes everywhere.